x-pack/plugins/actions/server/builtin_action_types/case/utils.ts

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { curry, flow } from 'lodash';
+import { schema } from '@kbn/config-schema';
+import { AxiosInstance, Method, AxiosResponse } from 'axios';
+
+import { ActionTypeExecutorOptions, ActionTypeExecutorResult, ActionType } from '../../types';
+
+import { ExecutorParamsSchema } from './schema';
+
+import {
+  CreateConnectorArgs,
+  ConnectorPublicConfigurationType,
+  CreateActionTypeArgs,
+  ExecutorParams,
+  MapRecord,
+  AnyParams,
+  CreateConnectorBasicArgs,
+  PrepareFieldsForTransformArgs,
+  PipedField,
+  TransformFieldsArgs,
+  Comment,
+  ExecutorActionParams,
+} from './types';
+
+import * as transformers from './transformers';
+
+import { SUPPORTED_SOURCE_FIELDS } from './constants';
+
+export const normalizeMapping = (supportedFields: string[], mapping: MapRecord[]): MapRecord[] => {
+  // Prevent prototype pollution and remove unsupported fields
+  return mapping.filter(
+    m => m.source !== '__proto__' && m.target !== '__proto__' && supportedFields.includes(m.source)
+  );
+};
+
+export const buildMap = (mapping: MapRecord[]): Map<string, MapRecord> => {
+  return normalizeMapping(SUPPORTED_SOURCE_FIELDS, mapping).reduce((fieldsMap, field) => {
+    const { source, target, actionType } = field;
+    fieldsMap.set(source, { target, actionType });
+    fieldsMap.set(target, { target: source, actionType });
+    return fieldsMap;
+  }, new Map());
+};
+
+export const mapParams = (
+  params: Partial<ExecutorActionParams>,
```

Comment:
Inside of `ExecutorActionParams` type remove `AnyParams` and do that and use `get` from lodash
```
export const mapParams = (
  params: Partial<ExecutorActionParams>,
  mapping: Map<string, MapRecord>
): AnyParams => {
  return Object.keys(params).reduce((prev: AnyParams, curr: string): AnyParams => {
    const field = mapping.get(curr);
    if (field) {
      prev[field.target] = get(params, curr);
    }
    return prev;
  }, {});
};
```